### DISCUSSION TITLE
Finding the top paths in GraphQL

### DISCUSSION THREAD

**Discussion by vasily-kartashov** [webonyx/graphql-php#1327] [Q&A]
Given a query like
```
query {
    customer {
        user {
           name
        }
        address {
           id
           country {
               name
           }
        }
    }
}
```

I would like to send the following increments to my datadog (i.e. all complete paths from root to leafs without the leafs):

```
increment('customer.user', 1)
increment('customer.address', 1)
increment('customer.address.country', 1)
```

How do I get such list based on the request and where do I add this tracing to my GraphQL?

**Comment by spawnia** [comment]
See https://github.com/nuwave/lighthouse/blob/5e360f38d33e29d5915acfaf6cb1b5208e36353b/src/Deprecation/DetectDeprecatedUsage.php for inspiration.
